tracer/src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -155,19 +179,32 @@ private static CreateTypeResult CreateProxyType(Type proxyDefinitionType, Type t
             //     }
             // }
 
-            lock (_locker)
+            lock (Locker)
             {
                 try
                 {
-                    var moduleBuilder = CreateTypeAndModuleBuilder(proxyDefinitionType, targetType, out var proxyTypeBuilder, out var instanceField);
+                    ModuleBuilder? moduleBuilder = null;
+                    TypeBuilder? proxyTypeBuilder = null;
+                    FieldInfo? instanceField = null;
+
+                    if (!dryRun)
+                    {
+                        moduleBuilder = CreateTypeAndModuleBuilder(proxyDefinitionType, targetType, out proxyTypeBuilder, out instanceField);
+                    }
 
                     if (proxyDefinitionType.IsValueType)
                     {
                         // Create Fields and Properties from the struct information
                         CreatePropertiesFromStruct(proxyTypeBuilder, proxyDefinitionType, targetType, instanceField);
 
+                        if (proxyTypeBuilder is null)
```

Comment:
Any reason this if statement couldn't be `if (dryRun)` ? It makes the condition slightly easier to grasp on a quick scan of the code

Reply:
no reasons, I made the change in [27c887d](https://github.com/DataDog/dd-trace-dotnet/pull/2716/commits/27c887de0ef4ed781468c1fb2ac05ba545e91964)